rust/autocomplete/src/config.rs

```diff
@@ -0,0 +1,67 @@
+use envconfig::Envconfig;
+use rdkafka::ClientConfig;
+
+#[derive(Envconfig, Clone)]
+pub struct Config {
+    #[envconfig(default = "postgres://posthog:posthog@localhost:5432/posthog")]
+    pub database_url: String,
+
+    #[envconfig(default = "10")]
+    pub max_pg_connections: u32,
+
+    #[envconfig(nested = true)]
+    pub kafka: KafkaConfig,
+
+    #[envconfig(default = "10")]
+    pub max_concurrent_transactions: usize,
+
+    #[envconfig(default = "10000")]
+    pub max_batch_size: usize,
+
+    // If a worker recieves a batch smaller than this, it will simply not commit the offset and
+    // sleep for a while, since DB ops/event scales inversely to batch size
+    #[envconfig(default = "1000")]
+    pub min_batch_size: usize,
+
+    #[envconfig(default = "100")]
+    pub next_event_wait_timeout_ms: u64,
+
+    #[envconfig(from = "BIND_HOST", default = "::")]
+    pub host: String,
+
+    #[envconfig(from = "BIND_PORT", default = "3301")]
+    pub port: u16,
+}
+
+#[derive(Envconfig, Clone)]
+pub struct KafkaConfig {
+    #[envconfig(default = "kafka:9092")]
+    pub kafka_hosts: String,
+    #[envconfig(default = "clickhouse_events_json")]
+    pub event_topic: String,
+    #[envconfig(default = "false")]
+    pub kafka_tls: bool,
+    #[envconfig(default = "false")]
+    pub verify_ssl_certificate: bool,
+    #[envconfig(default = "autocomplete-rs")]
+    pub consumer_group: String,
+}
+
+impl From<&KafkaConfig> for ClientConfig {
+    fn from(config: &KafkaConfig) -> Self {
+        let mut client_config = ClientConfig::new();
+        client_config
+            .set("bootstrap.servers", &config.kafka_hosts)
+            .set("statistics.interval.ms", "10000")
+            .set("group.id", config.consumer_group.clone())
+            .set("enable.auto.offset.store", "false"); // We store on a per-message basis anyway right now, but this is for later if we decide to work in batches
```

Comment:
I'm not so sure about this line. The offset store is just the memory store of the last offsets seen. `auto.commit` is what commits offsets, and it's still on. But I'm pretty sure it needs the offset store?

Either way, I don't see us committing offsets anywhere, so I think we want to use auto.commit and therefore want to remove this line and leave the offset store on?

(Probably not important for the code as it is now, but I'd rather not forget it for prod.)

Reply:
You're right, if we fail to make forward progress we bail out of the loop and fall over, we shouldn't handle any offset committing manually at all (it's fine if falling over causes some messages to be re-processed)